OtripleS.Portal.Web.Tests.Unit/Services/Courses/CourseServiceTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net.Http;
+using Moq;
+using OtripleS.Portal.Web.Brokers.Apis;
+using OtripleS.Portal.Web.Brokers.Logging;
+using OtripleS.Portal.Web.Models.Courses;
+using OtripleS.Portal.Web.Services.Courses;
+using RESTFulSense.Exceptions;
+using Tynamix.ObjectFiller;
+using Xeptions;
+using Xunit;
+
+namespace OtripleS.Portal.Web.Tests.Unit.Services.Courses
+{
+    public partial class CourseServiceTests
+    {
+        private readonly Mock<IApiBroker> apiBrokerMock;
+        private readonly Mock<ILoggingBroker> loggingBrokerMock;
+        private readonly ICourseService courseService;
+
+        public CourseServiceTests()
+        {
+            this.apiBrokerMock = new Mock<IApiBroker>();
+            this.loggingBrokerMock = new Mock<ILoggingBroker>();
+
+            this.courseService = new CourseService(
+                apiBroker: this.apiBrokerMock.Object,
+                loggingBroker: this.loggingBrokerMock.Object);
+        }
+
+        public static TheoryData CriticalApiExceptions()
+        {
+            string exceptionMessage = GetRandomString();
+            var responseMessage = new HttpResponseMessage();
+
+            var httpRequestException =
+                new HttpRequestException();
+
+            var httpResponseUrlNotFoundException =
+                new HttpResponseUrlNotFoundException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            var httpResponseUnAuthorizedException =
+                new HttpResponseUnauthorizedException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            return new TheoryData<Exception>
+            {
+                httpRequestException,
+                httpResponseUrlNotFoundException,
+                httpResponseUnAuthorizedException
+            };
+        }
+
+        public static TheoryData DependencyApiExceptions()
+        {
+            string exceptionMessage = GetRandomString();
```

Comment:
I would argue that it's pretty clean that we're getting a string based on the right side. Okay as is, not a big deal I guess :)